group_processor.go

```
@@ -91,6 +91,40 @@ func (gp *GroupProcessor) New() (err error) {
 	return nil
 }
 
+func (gp *GroupProcessor) SetOffsets(offsets map[int32]int64) {
+	gp.RLock()
+	defer gp.RUnlock()
+
+	for partition, offset := range offsets {
+		oldest, err := gp.kafka.client.GetOffset(gp.Topic, partition, sarama.OffsetOldest)
+
+		if err != nil {
+			gp.log.Panicf(err, "Couldn't get offsets for %s/%d. Resuming from default offset", gp.Topic, partition)
```

Maybe it would be better to return the error two cases where the offsets can't be retrieved?